metadata-ingestion/src/datahub/ingestion/source/couchbase/couchbase_kv_schema.py

```diff
@@ -0,0 +1,170 @@
+from collections import Counter
+from typing import Any, Counter as CounterType, Dict, Iterable, List, Tuple, Union
+
+from typing_extensions import TypedDict
+
+
+class BasicSchemaDescription(TypedDict):
+    types: CounterType[Union[type, str]]  # field types and times seen
+    count: int  # times the field was seen
+
+
+class SchemaDescription(BasicSchemaDescription):
+    delimited_name: str  # collapsed field name
+    # we use 'mixed' to denote mixed types, so we need a str here
+    type: Union[type, str]  # collapsed type
+    nullable: bool  # if field is ever missing
+
+
+class CouchbaseFieldData:
+    types: Tuple[str, ...] = ()
+    samples: List[Any] = []
+
+
+def json_schema(
```

Comment:
we already have a number of utilities for doing "schema inference" from objects - e.g. https://github.com/datahub-project/datahub/blob/0361f2463db5f64dead3ea9b24af853498bbf28f/metadata-ingestion/src/datahub/ingestion/source/schema_inference/object.py#L86

Does this stuff still make sense?